defaults/main.yml

```diff
@@ -1,30 +1,31 @@
 ---
 # defaults file for ansible-role-openio-keepalived
 
-keepalived_conf_template: keepalived.conf.j2
 keepalived_conf_path: /etc/keepalived/keepalived.conf
 
 keepalived_global_defs_notification_email: support@openio.io
 keepalived_global_defs_notification_email_from: "{{ inventory_hostname }}@openio.io"
 keepalived_global_defs_smtp_server: 127.0.0.1
 keepalived_global_defs_smtp_connect_timeout: 30
+keepalived_track_script: check_haproxy
+keepalived_auth_type: PASS
 
 keepalived_vrrp_instances:
   VI_01:
-    state: MASTER
-    interface: "{{ ansible_default_ipv4.interface }}"
-    virtual_router_id: '51'
-    priority: '100'
-    advert_int: '1'
+    state: "{{ instance_state|default('BACKUP') }}"
+    interface: "{{ instance_interface }}"
+    virtual_router_id: "{{ instance_vrid|default('51') }}"
+    priority: "{{ instance_priority|default('100') }}"
+    advert_int: "{{ instance_advert_int|default('1') }}"
     authentication:
-      auth_type: PASS
-      auth_pass: AUTH_PASS
+      auth_type: "{{ keepalived_auth_type }}"
+      auth_pass: "{{ instance_pwd }}"
     virtual_ipaddress:
-      - 172.17.1.2
+      - "{{ instance_vip }} dev {{ instance_interface }}"
     track_script:
-      - check_haproxy
+      - "{{ keepalived_track_script }}"
     track_interface:
-      - "{{ ansible_default_ipv4.interface }}"
-
-keepalived_provision_only: false
+      - "{{ instance_interface }}"
+    unicast_src_ip: "{{ instance_unicast_src_ip }}"
+    unicast_peer: "{{ instance_unicast_peer_ips }}"
```

Comment:
Adding a default here to `"{{ groups['keepalived'] | map('extract', hostvars, ['openio_bind_address']) | list }}"` would match our global default

Reply:
Shouldn't `unicast_peer` mean the `keepalived` group excluding the server with `unicast_src_ip`? That's why I didn't default it to anything and wrote a task to pick up the right IPs.

Reply:
It works with the local IP in the list but it's nice to remove it (you can achieve it using the difference filter I suppose).

Reply:
Yes, that's what I'm using. So can we agree to keep this variable since it's too much to have something like `unicast_peer: "{{ instance_unicast_peer_ips | default(something) }}"` here?